cmd/antrea-agent/agent.go

```diff
@@ -458,8 +458,12 @@ func run(o *Options) error {
 	}
 
 	if features.DefaultFeatureGate.Enabled(features.Multicast) {
-		mcastController := multicast.NewMulticastController(ofClient, nodeConfig, ifaceStore)
-		if err := mcastController.Initialize(); err != nil {
+		mcastController, err := multicast.NewMulticastController(ofClient, nodeConfig, ifaceStore, ovsBridgeClient)
+
+		if err != nil {
+			return fmt.Errorf("error creating multicast controller: %s", err.Error())
+		}
+		if err := mcastController.Initialize(o.config.ExternalMulticastInterfaces, networkConfig.TransportIface); err != nil {
```

Comment:
I guess it is because these two are used only once in Initialize(). If we follow this way, maybe ovsBridgeClient can be an argument of Initialize() too.

Reply:
Updated. Yes, ovsBridgeClient is also used in Initialize() only so I move it as a parameter of Initialize() as well.

Reply:
OK, I feel it's more clear to inject all dependencies in constructor, regardless of whether they are only used once or multiple times.